### PR TITLE
API Make use of the new AdminController

### PR DESCRIPTION
**Pull request by GuySartorelli**
Make `CMSExternalLinksController` a subclass of `AdminController` introduced in https://github.com/silverstripe/silverstripe-admin/pull/1836

## Issue
- https://github.com/silverstripe/silverstripe-admin/issues/1761

### CODE REVIEW

File: src/Controllers/CMSExternalLinksController.php [hunk 1]
```diff
@@ -2,63 +2,54 @@
...
         if ($track) {
-            return json_encode([
+            return $this->jsonSuccess(200, [
                 'TrackID' => $track->ID,
                 'Status' => $track->Status,
                 'Completed' => $track->getCompletedPages(),
                 'Total' => $track->getTotalPages()
             ]);
         }
+        return $this->jsonSuccess(200, []);
```
Comment (GuySartorelli) [inline src/Controllers/CMSExternalLinksController.php:41]: Returning nothing (by returning `$this->jsonSuccess(200);`) resulted in a "parsererror" toast message. So I had to include the `[]` there.
Looks like the toast was being triggered by something upstream of the js in this module itself, so I couldn't just solve it there.

File: src/Controllers/CMSExternalLinksController.php [hunk 2]
```diff
@@ -2,63 +2,54 @@
...
      * Respond to Ajax requests for info on a running job
-     *
-     * @return string JSON string detailing status of the job
      */
-    public function getJobStatus()
+    public function getJobStatus(): HTTPResponse
     {
-        if (!Permission::check('CMS_ACCESS_CMSMain')) {
-            return $this->httpError(403, 'You do not have permission to access this resource');
-        }
```
Comment (GuySartorelli) [inline src/Controllers/CMSExternalLinksController.php:30]: Handled by `AdminController::init()`

File: src/Controllers/CMSExternalLinksController.php [hunk 3]
```diff
@@ -2,63 +2,54 @@
...
-        // Set headers
-        HTTPCacheControlMiddleware::singleton()->setMaxAge(0);
```
Comment (GuySartorelli) [inline src/Controllers/CMSExternalLinksController.php:32]: This middleware is disabled in `AdminController::init()`

File: src/Controllers/CMSExternalLinksController.php [hunk 4]
```diff
@@ -2,63 +2,54 @@
...
-        $this->response
-            ->addHeader('Content-Type', 'application/json')
-            ->addHeader('Content-Encoding', 'UTF-8')
```
Comment (GuySartorelli) [inline src/Controllers/CMSExternalLinksController.php:35]: UTF-8 encoding is either not needed anymore or is done by default - works without that line.
The content type is set by `AdminController::jsonSuccess()`

File: client/src/js/BrokenExternalLinksReport.js [hunk 5]
```diff
@@ -99,7 +99,7 @@
           async: true,
           success(data) {
             // No report, so let user create one
-            if (!data) {
+            if (!data || (typeof data === 'object' && data.length < 1)) {
```
Comment (GuySartorelli) [inline client/src/js/BrokenExternalLinksReport.js:102]: no data is no option anymore, but I kept it there as defensive programming.

File: _config/routes.yml
Comment (GuySartorelli) [inline _config/routes.yml:1]: Yay we don't need this silly routing rule anymore

File: src/Controllers/CMSExternalLinksController.php [hunk 6]
```diff
@@ -2,63 +2,54 @@
...
-use SilverStripe\Security\Permission;
 
-class CMSExternalLinksController extends Controller
+class CMSExternalLinksController extends AdminController
 {
+    private static ?string $url_segment = 'externallinks';
+
+    private static string|array $required_permission_codes = [
+        'CMS_ACCESS_CMSMain',
+    ];
```
Comment (GuySartorelli) [inline src/Controllers/CMSExternalLinksController.php:19]: Taken from the explicit permission checks that used to be in the actions.

File: src/Controllers/CMSExternalLinksController.php (hunk 1 above)
Comment (GuySartorelli) [reply src/Controllers/CMSExternalLinksController.php:41]: Same thing with `start()` below.

**Review by emteknetnz — APPROVED** [review]

**Pull request merged**